src/passes/Print.cpp

```diff
@@ -380,6 +380,39 @@ struct PrintSExpression : public UnifiedExpressionVisitor<PrintSExpression> {
       visitExpression(curr);
     }
   }
+  void visitContNew(ContNew* curr) {
+    if (!maybePrintUnreachableReplacement(curr, curr->type)) {
+      visitExpression(curr);
+    }
+  }
+  void visitContBind(ContBind* curr) {
+    if (!maybePrintUnreachableOrNullReplacement(
+          curr, Type(curr->sourceType, Nullable)) ||
```

Comment:
I don't understand the rationale behind axing `sourceType` and `contType`. They represent immediates on the respective instructions. The immediates may point to types that differ from the types of the operands (in these cases `curr->cont`).

Reply:
The idea is that by recomputing the type immediates when necessary rather than caching them, we always produce the most precise possible types. If an optimization refines the type of the input to one of these instructions, then refinalization will refine the output type automatically. That couldn't happen if refinalization had to respect the cached type immediates.